CMakeLists.txt

```diff
@@ -137,10 +119,35 @@ target_include_directories(tlsuv
         ${CMAKE_CURRENT_SOURCE_DIR}/src
         )
 
+target_compile_definitions(tlsuv PRIVATE
+        TLSUV_VERSION=v${PROJECT_VERSION}
+        USE_$<UPPER_CASE:${TLSUV_TLSLIB}>
+        TLS_IMPL=${TLSUV_TLSLIB}
+)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    target_compile_definitions(tlsuv PRIVATE _POSIX_C_SOURCE=200112 _GNU_SOURCE)
```

Comment:
Are these really for "Linux", or "gcc"? Similarly are the definitions in the "WIN32" section for WIN32 or MSVC?